backend/src/main/kotlin/com/tngtech/apicenter/backend/connector/database/service/SpecificationDatabaseService.kt

```diff
@@ -69,4 +64,39 @@ class SpecificationDatabaseService constructor(
             throw VersionAlreadyExistsException(specificationEntity.title)
         }
     }
+
+    override fun saveOne(version: Version, serviceId: ServiceId, fileUrl: String?) {
+        val existingSpec = specificationRepository.findById(serviceId.id)
+
+        if (!existingSpec.isPresent) {
+            val newSpecification = Specification(
+                    serviceId,
+                    version.metadata.title,
+                    version.metadata.description,
+                    listOf(version),
+                    fileUrl
+            )
+            val newSpecificationEntity = specificationEntityMapper.fromDomain(newSpecification)
+            mapAndStoreEntity(newSpecificationEntity)
+        }
+
+        existingSpec.ifPresent { s ->
+            val versionStrings = s.versions.map { v -> versionEntityMapper.toDomain(v).metadata.version }
+            val newVersionString = version.metadata.version
+            if (versionStrings.contains(newVersionString)) {
+
+                val existingContents = s.versions.firstOrNull {
+                    versionEntity -> versionEntityMapper.toDomain(versionEntity).metadata.version == newVersionString
+                }?.content
+
+                if (existingContents == version.content) {
```

Comment:
I'd like to do this check in the `domain` package, actually. This is not a characteristic of the DB (or another persistence layer), but of ApiCenter's domain logic.